File: framework/languages/dotnet/samples/SupportChat/Server/Support/Infrastructure/ZLink/Actors/SupportUserActorFactory.cs
using Zlink.Framework.Contracts.Actors;

namespace SupportChat.Server.Support.Infrastructure.ZLink.Actors;

internal sealed class SupportUserActorFactory : IZLinkActorFactory
{
    public ValueTask<IZLinkActor> CreateAsync(
        string actorId,
        IZLinkActorContext context,
        CancellationToken cancellationToken = default)
    {
        return ValueTask.FromResult<IZLinkActor>(new SupportUserActor(actorId, context));
    }
}

File: framework/languages/node/samples/SupportChat.Ts/Server/Support/Infrastructure/ZLink/Actors/support-user-actor-factory.ts
import { Injectable } from '@nestjs/common';
import { SupportUserActor } from './support-user-actor';
import type { ZLinkActorContext, ZLinkActorFactory } from '@zlink-systems/framework';

@Injectable()
class SupportUserActorFactory implements ZLinkActorFactory {
  async create(actorId: string, context: ZLinkActorContext): Promise<SupportUserActor> {
    return new SupportUserActor(actorId, context);
  }
}

export { SupportUserActorFactory };
